test/functional/test_framework/mininode.py

```diff
@@ -585,7 +585,7 @@ def send_blocks_and_test(self, blocks, node, *, success=True, force_send=False,
                     self.send_message(msg_block(block=b))
             else:
                 self.send_message(msg_headers([CBlockHeader(block) for block in blocks]))
-                wait_until(lambda: blocks[-1].sha256 in self.getdata_requests, timeout=timeout, lock=mininode_lock)
+                wait_until(lambda: blocks[-1].sha256 in self.getdata_requests, timeout=timeout * self.options.factor, lock=mininode_lock)
```

Comment:
Given that they are all based on `wait_until` wouldn't it make more sense to treat `wait_until` as the basic mechanism for timed waits and only multiply it in there?

Reply:
Meanwhile, I have changed a few other things, because I faced various p2p errors (the P2PInterface didn't have their factor variable, so I had to redo a few things). And yes, wait_until will be changed too.